packages/react-component-library/src/components/Table/Table.tsx

```diff
@@ -61,6 +75,7 @@ export const Table: React.FC<TableProps> = ({
         <tbody>
           {tableData.map((row: RowProps) => (
             <StyledTableRow
+              onClick={() => handleRowClick(row.id)}
```

Comment:
Might be able to do this inline with optional chaining `onRowClick?(id)`.

Fine as is though.